doc/dev/osd_internals/manifest.rst

```diff
@@ -46,24 +48,31 @@ There are some rough edges we may want to improve:
     do_osd_ops.  This probably isn't desirable, see Snapshots section
     below for some options on how generally to mix these operations
     with snapshots.
+    --> Need to fix
   * Appears to assume that the corresponding section of the object
     does not exist (sets FLAG_MISSING) but does not check whether the
     corresponding extent exists already in the object.
+    --> Is this a problem ? There is a workaround (https://github.com/ceph/ceph/pull/28591)
```

Comment:
Yeah, I think it's a problem.  I don't think it should be possible for a rados operation to break a core invariant of the manifest metadata -- object regions marked missing must be missing.

Reply:
If that's not an invariant, it might be good to add some explanation of what the rules are for the manifest metadata.  What *must* be true of the relationship between an object and the manifest?